cmd/mtaBuild.go

```diff
@@ -261,65 +254,95 @@ func runMtaBuild(config mtaBuildOptions,
+	}
 
-				config.MtaDeploymentRepositoryURL += config.MtarGroup + "/" + mtarArtifactName + "/" + config.Version + "/" + fmt.Sprintf("%v-%v.%v", mtarArtifactName, config.Version, "mtar")
+	credentialsEncoded := "Basic " + base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", config.MtaDeploymentRepositoryUser, config.MtaDeploymentRepositoryPassword)))
+	headers := http.Header{}
+	headers.Add("Authorization", credentialsEncoded)
 
-				commonPipelineEnvironment.custom.mtarPublishedURL = config.MtaDeploymentRepositoryURL
+	config.MtarGroup = strings.ReplaceAll(config.MtarGroup, ".", "/")
+	mtarArtifactName := mtarName
+	if !isMtarNativelySuffixed {
+		mtarArtifactName = strings.TrimSuffix(mtarArtifactName, ".mtar")
+	}
 
-				log.Entry().Infof("pushing mtar artifact to repository : %s", config.MtaDeploymentRepositoryURL)
+	config.MtaDeploymentRepositoryURL += config.MtarGroup + "/" + mtarArtifactName + "/" + config.Version + "/" + fmt.Sprintf("%v-%v.%v", mtarArtifactName, config.Version, "mtar")
+	commonPipelineEnvironment.custom.mtarPublishedURL = config.MtaDeploymentRepositoryURL
 
-				data, err := os.Open(getMtarFilePath(config, mtarName))
-				if err != nil {
-					return errors.Wrap(err, "failed to open mtar archive for upload")
-				}
-				_, httpErr := httpClient.SendRequest("PUT", config.MtaDeploymentRepositoryURL, data, headers, nil)
+	log.Entry().Infof("pushing mtar artifact to repository : %s", config.MtaDeploymentRepositoryURL)
 
-				if httpErr != nil {
-					return errors.Wrap(err, "failed to upload mtar to repository")
-				}
-			} else {
-				return errors.New("mtarGroup, version not found and must be present")
+	mtarPath := getMtarFilePath(config, mtarName)
+	data, err := utils.Open(mtarPath)
+	if err != nil {
+		return errors.Wrap(err, "failed to open mtar archive for upload")
+	}
+	defer data.Close()
 
-			}
+	if _, httpErr := utils.SendRequest("PUT", config.MtaDeploymentRepositoryURL, data, headers, nil); httpErr != nil {
+		return errors.Wrap(httpErr, "failed to upload mtar to repository")
+	}
 
-		} else {
-			return errors.New("mtaDeploymentRepositoryUser, mtaDeploymentRepositoryPassword and mtaDeploymentRepositoryURL not found , must be present")
+	if config.CreateBuildArtifactsMetadata {
+		if err := buildArtifactsMetadata(config, commonPipelineEnvironment, mtarPath); err != nil {
+			log.Entry().Warnf("unable to create build artifacts metadata: %v", err)
+			return nil
 		}
-	} else {
-		log.Entry().Infof("no publish detected, skipping upload of mtar artifact")
 	}
-	return err
+
+	return nil
+}
+
+func buildArtifactsMetadata(config mtaBuildOptions, commonPipelineEnvironment *mtaBuildCommonPipelineEnvironment, mtarPath string) error {
+	mtarDir := filepath.Dir(mtarPath)
+	buildArtifacts := build.BuildArtifacts{
+		Coordinates: []versioning.Coordinates{
+			{
+				GroupID:    config.MtarGroup,
+				ArtifactID: config.MtarName,
+				Version:    config.Version,
+				Packaging:  "mtar",
+				BuildPath:  mtarDir,
```

Comment:
The build path should be path of mta.yaml file and not the mtar. Is mtar generated at the same directory as of mta.yaml? or users can choose to change the path?

Reply:
[318a082](https://github.com/SAP/jenkins-library/pull/5166/commits/318a082da61b3bef69b790eebdff3f2394fa81cf)